packages/gatsby/src/utils/cache.js

```diff
@@ -1,73 +1,44 @@
-const Promise = require(`bluebird`)
 const fs = require(`fs-extra`)
-const _ = require(`lodash`)
-
-const objectToMap = obj => new Map(Object.entries(obj))
+const manager = require(`cache-manager`)
+const fsStore = require(`cache-manager-fs`)
+const path = require(`path`)
+
+const MAX_CACHE_SIZE = 250
+
+class Cache {
+  constructor(folderName = `db`, store = fsStore) {
+    this.folderName = folderName
+
+    const caches = [
+      { store: `memory`, max: MAX_CACHE_SIZE },
```

Comment:
Ideally, there'd probably be some way to opt into this at the plugin level.

Reply:
just so it doesn't get lost - `max` is number of documents hold in memory with default config settings, but we can use additional `length` config option that could assign "weights" to documents and use some kind of heuristic to somewhat be able to limit size of in-memory cache in actual bytes to keep this under control

Reply:
@pieh you think it's worth experimenting with that in this PR still, or should we go ahead and merge this today sometime?